game/src/core/utils/files/FileSystemUtil.java

```diff
@@ -0,0 +1,144 @@
+    //noinspection InstantiationOfUtilityClass
+    final FileSystemUtil helper = new FileSystemUtil();
+    try {
+      return Objects.requireNonNull(
+              helper.getClass().getResource(helper.getClass().getSimpleName() + ".class"))
+          .toURI()
+          .getScheme()
+          .equals("jar");
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static boolean isStartedInJUnitTest() {
+    for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
+      if (element.getClassName().startsWith("org.junit.")) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private static URI getUriToJarFileEntry() {
+    //noinspection InstantiationOfUtilityClass
+    final FileSystemUtil helper = new FileSystemUtil();
+    try {
+      return new URI(
+          Objects.requireNonNull(
+              Objects.requireNonNull(
+                      helper.getClass().getResource(helper.getClass().getSimpleName() + ".class"))
+                  .toURI()
+                  .toURL()
+                  .toExternalForm()));
+    } catch (URISyntaxException | MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * This method searches for all asset files in the subdirectories of the specified directory. It
+   * is not necessary to explicitly specify the asset storage location, as this method automatically
+   * determines the storage location and how the game was started. Furthermore, a {@link
+   * SimpleFileVisitor} must be passed, which is called for each file or folder found. You can then
+   * specify yourself what should happen to the files found, for example mapping them into another
+   * structure or reading them to use them in the game.
+   *
+   * @param pathToDirectory is a {@link IPath} relative to the root asset directory of the game to
+   *     search.
+   * @param visitor is a {@link SimpleFileVisitor} for {@link Path}, which is called for each file
+   *     or folder found.
+   */
+  public static void searchAssetFilesInSubdirectories(
+      final IPath pathToDirectory, final SimpleFileVisitor<Path> visitor) {
+    if (isStartedInJUnitTest()) {
+      // inside JUnit test
+      try {
+        Files.walkFileTree(
+            Paths.get(
+                Objects.requireNonNull(
+                        Thread.currentThread()
+                            .getContextClassLoader()
+                            .getResource(pathToDirectory.pathString()))
+                    .toURI()
+                    .normalize()),
+            visitor);
+      } catch (IOException | URISyntaxException e) {
+        throw new RuntimeException(e);
+      }
+    } else if (isStartedInJarFile()) {
+      // inside JAR file
+      try (FileSystem fileSystem =
+          FileSystems.newFileSystem(getUriToJarFileEntry(), Collections.emptyMap())) {
+        Files.walkFileTree(fileSystem.getPath(pathToDirectory.pathString()), visitor);
```

Comment:
hier tritt ein fehler auf, auch wenn das spiel mit `./gradlew dungeon:runStarter` gestartet wurde (also gar nicht innerhalb eines jar files).

Reply:
schräg. mir ist aufgefallen, dass gradle für game immer ein jar anlegt, insbesondere zum anlegen des starter.jar geht das nur, wenn es auch ein game.jar gibt. ich glaube, die verwendeten konstrukte in der build.gradle erzwingen das (game ist eine "java-lib", und der jar-task in dungeon:build.gradle ist seltsam umständlich formuliert)?